frontend/src/components/worksheets/NewUpload/NewUpload.jsx

```diff
@@ -171,6 +172,18 @@ class NewUpload extends React.Component<{
         this.setState({ percentComplete: 0, uploading: false });
     }
 
+    // Handling cancelling
+    // Based on: https://stackoverflow.com/questions/49627109/reactjs-how-to-detect-when-cancel-is-clicked-on-file-input

```

Comment:
Nit: I'm not sure if we need to include the StackOverflow link here. I think just a brief description of the fix should suffice.